### PR TITLE
Scoped variants

### LINKED ISSUE #2053: group_if, group_at, group_all
`group_by(everything())` doesn't work, probably because `group_by()` has `mutate()` semantics.

``` r
group_by_all <- function(.data) group_by_(.data, .dots = colnames(.data))
```

This is different from `rowwise()` iff the data contains duplicate rows.

### LINKED ISSUE #2106: transmute_at, transmute_if, transmute_all

### LINKED ISSUE #2120: rename_all, rename_if, rename_at
When reading CSV or Excel files, it's very common to change all column names to lowercase or apply to them some transformation (some `sub`, `gsub`, `str_replace`...).

It's possible to use the traditional syntax:

``` r
colnames(df1) <- tolower(colnames(df1))
```

It would be desirable that dplyr could have a function for it.
I have just discovered this:

``` r
df1 <- df1 %>% 
  `names<-`(tolower(names(.)))
```

( from here https://github.com/romainfrancois/budflights/blob/master/gen/gen.R )

But wouldn't it be better to have an specific function with a better syntax (more legible)?

### LINKED ISSUE #2417: filter_if, filter_at, filter_all
i.e.

```R
filter_all(.tbl, .funs, ...)
filter_if(.tbl, .predicate, .funs,  ...)
filter_at(.tbl, .cols, .funs, ...)
```

### PR DESCRIPTION
**Pull request by lionel-**
Scoped variants for:

* `filter()`, closes #2417
* `group_by()`, closes #2053
* `rename()` and `select()`, closes #2120
* `transmute()`, closes #2106

Arguments expecting functions are now more flexible:

* The `.predicate` arguments (for `_if` variants) are passed to `as_function()` (depends on https://github.com/hadley/rlang/pull/77).
* The `.funs` arguments now allow quoted functions (name, quosure) directly (before these had to be wrapped in `funs()`.
* `.funs` is passed both by value and by expression so that named functions can be handled by hybrid evaluation without sacrificing value semantics.
* Anonymous functions are also supported so that `funs()` supports unquoting and `.funs` can be passed a function-valued expression (like `summarise_all(df, identity(mean))`). This has the downside that there's no hybrid evaluation for unnamed functions.

### CODE REVIEW

**Comment by lionel-** [comment]
I wonder if `quo_name()` applied to a literal should return the sexp type. We currently abbreviate deparsed code:

```r
mutate_all(data_frame(1:3), funs(!! mean, !! sd))
#> # A tibble: 3 × 3
#>   `1:3` `function (x, ...) ...` `function (x, na.rm = FALSE) ...`
#>   <int>                   <dbl>                             <dbl>
#> 1     1                       2                                 1
#> 2     2                       2                                 1
#> 3     3                       2                                 1
```

File: R/colwise-filter.R [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#' # Or the union:
+#' filter_all(mtcars, any_vars(. > 150))
+#'
+#'
+#' # You can vary the selection of columns on which to apply the
+#' # predicate. filter_at() takes a vars() specification:
+#' filter_at(mtcars, vars(starts_with("d")), any_vars((. %% 2) == 0))
+#'
+#' # And filter_if() selects variables with a predicate function:
+#' filter_if(mtcars, rlang::is_integerish, all_vars(. != 0))
```
Comment (hadley) [inline R/colwise-filter.R]: Can you just use `is.integer()` here?

File: R/colwise-group-by.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#' @inheritParams scoped
+#' @param .add Passed to the `add` argument of [group_by()].
+#' @export
+#' @examples
+#' # Group a data frame by all variables:
+#' group_by_all(mtcars)
+#'
+#' # Group by variables selected with a predicate:
+#' group_by_if(iris, is.factor)
+#' group_by_if(mtcars, rlang::is_integerish)
```
Comment (hadley) [inline R/colwise-group-by.R]: And here

File: NEWS.md [hunk 3]
```diff
@@ -16,6 +22,23 @@
 
 * `funs()` has better handling of namespaced functions (#2089).
 
+* `rename_vars()` gains a `strict` argument to control if an
+  error is thrown when you try and rename a variable that doesn't
+  exist.
+
+* You can create a `tbl` directly from a DBI connection (#2576).
```
Comment (hadley) [inline NEWS.md]: Can you delete these three bullets?

**Pull request merged**